scipy/stats/_stats_py.py

```diff
@@ -1013,6 +1013,16 @@ def _moment(a, moment, axis, *, mean=None):
         # Starting point for exponentiation by squares
         mean = a.mean(axis, keepdims=True) if mean is None else mean
         a_zero_mean = a - mean
+
+        eps = np.finfo(a_zero_mean.dtype).resolution * 1e6
+        rel_diff = np.max(np.abs(a_zero_mean), axis=axis, keepdims=True)/mean
+        with np.errstate(invalid='ignore'):
+            precision_loss = np.any((0 < rel_diff) & (rel_diff < eps))
+        if precision_loss:
+            message = ("Precision loss occurred in moment calculation due to "
+                       "catastrophc cancellation. Results may be unreliable.")
```

Comment:
I think the message is just missing an information for the user. Is this actionable or not for them. If you get the warning, then what? I would add a line in that sense.

Reply:
Maybe we can recommend to play with dtype or say in which condition this usually happen?

Reply:
Better yes 👍

Reply:
Committed. Thanks! That will be helpful.